### PR TITLE
Google Drive step3 indexing not starting bug fix

### PR DESCRIPTION
**Pull request by meherhendi**
This code fixes a bug with Google Drive where Step 3 don't appear, and it's impossible to index files from Google Drive,

this is caused by the find loop not being able to find the Gdrive credentials from the credentials array due to the line 
`credential.credential_json?.google_drive_tokens && credential.user_id === null` as none of the credentials matches this condition 

this fix will pull the current user credentials and matches his ID to provide him with the right credentials

### CODE REVIEW

File: web/src/app/admin/connectors/google-drive/page.tsx [hunk 1]
```diff
@@ -317,12 +329,21 @@ const Main = () => {
...
+      </div>
+    );
+  }
   const googleDrivePublicCredential:
     | Credential<GoogleDriveCredentialJson>
     | undefined = credentialsData.find(
     (credential) =>
       credential.credential_json?.google_drive_tokens &&
-      credential.user_id === null
+      credential.user_id === currentUserData?.id
```
Comment (Weves) [inline web/src/app/admin/connectors/google-drive/page.tsx]: Another way of doing this that imo better replicates the existing logic is to change this check to `credential.credential_json?.google_drive_tokens && credential.is_admin`. 

For now, we generally assume that there should/could only be one live admin credential. This keeps the logic here simpler for now, and enables multiple admins to see / modify each others connectors.

Let me know if that makes sense / if you have questions about that!

**Review by Weves — CHANGES_REQUESTED** [review]
Hey @meherhendi! Thanks a bunch for this fix, and sorry about the slow turnaround on review. 

I added a small comment about a different possible approach to fixing this. Let me know what you think, happy to discuss more!

File: web/src/app/admin/connectors/google-drive/page.tsx (hunk 1 above)
Comment (meherhendi) [reply web/src/app/admin/connectors/google-drive/page.tsx]: Totally agree, better to keep the logic simple atm

File: web/src/app/admin/connectors/google-drive/page.tsx [hunk 2]
```diff
@@ -276,13 +277,24 @@ const Main = () => {
     refreshCredentials,
   } = usePublicCredentials();
 
+  const {
```
Comment (Weves) [inline web/src/app/admin/connectors/google-drive/page.tsx]: Since we're no longer actively using the `user` below, can we remove this call?

Comment (meherhendi) [reply web/src/app/admin/connectors/google-drive/page.tsx]: Absolutely, 
PS: kept  `user_id: string | null` as the id is returned as a string

**Review by Weves — APPROVED** [review]

**Comment by Weves** [comment]
Thanks a bunch for this fix!

**Pull request merged**